docs(lane_change): object filtering description (#9947)
autowarefoundation/autoware.universe: planning/behavior_path_planner/autoware_behavior_path_lane_change_module/README.md

lateral_acceleration_resolution = (maximum_lateral_acceleration - minimum_lateral_acceleration) / lateral_acceleration_sampling_num
```

> ##### Object filtering

Reviewer: ```suggestion
#### Object filtering
```

  #LightGreen:To SUBPROCESS "Separate object based on its behavior in target lane";
  if(can separate object in SUBPROCESS) then (TRUE)
  stop
  endif
  else (FALSE)
  if (Object is moving\nAND\nis a vehicle class\nAND\npath overlaps target lane?) then (TRUE)
  #LightGreen:To SUBPROCESS "Separate object based on its behavior in target lane";
  if(can separate object in SUBPROCESS) then (TRUE)
  stop
  endif
  endif
  endif
  endif
  #LightPink:From SUBPROCESS "Separate object based on its behavior in target lane";
  if (Object is in expanded target lane\nAND\nis stopped\nAND\nin front of ego?) then (TRUE)
  :Add object to <color:blue>**filtered_object.leading_objects.stopped_at_bound**</color> list;
  stop
  endif
  else (FALSE)
  if (Object overlaps preceding target lanes?) then (TRUE)
  :Add object to <color:blue>**filtered_object.trailing_objects**</color> list;
  stop
  endif
endif
}
if (Object is in front of ego\nAND\nis before terminal\nAND\noverlaps current lane?) then (TRUE)
  :Add object to <color:orange>**filtered_object.current_lane**</color> list;
  stop
else (FALSE)
  :Add object to <color:magenta>**filtered_object.others**</color> list;
endif

stop

> #LightGreen:Start separating object based on its behavior in target lane;
> if (Object is behind ego AND moving?) then (TRUE)
>   :Add object to <color:blue>**filtered_object.trailing_objects**</color> list;
>   :Can separate object;
>   stop
> else (FALSE)
>   if (Object is in front of ego?) then (TRUE)
>     if (Object is moving?) then (TRUE)
>     :Add object to <color:blue>**filtered_object.leading_objects.moving**</color> list;
>     :Can separate object;
>     stop
>   else (FALSE)
>   :Add object to <color:blue>**filtered_object.leading_objects.stopped**</color> list;
>   :Can separate object;
>   stop
>   endif
>   endif
> endif
> #LightPink:Cant separate proceed with other checks;

Reviewer: ```suggestion
group Target Lane Objects Subprocess #LightYellow
start
if (Object is behind ego?) then (TRUE)
  if (Object is moving?) then (TRUE)
    :Add object to <color:blue>**filtered_object.trailing_objects**</color> list;
    #LightGreen:Can separate object;
    stop
  else (FALSE)
    #LightPink:Cant separate proceed with other checks;
    stop
  endif
else (FALSE)
  if (Object is moving?) then (TRUE)
    :Add object to <color:blue>**filtered_object.leading_objects.moving**</color> list;
  else (FALSE)
    :Add object to <color:blue>**filtered_object.leading_objects.stopped**</color> list;
  endif
  #LightGreen:Can separate object;
  stop
endif
```
![target_lane_objects_subprocess](https://github.com/user-attachments/assets/b48cc112-62de-49e2-be57-b63357e98936)
